frontend/src/pages/functional_components/NewExperimentFC.tsx

```diff
@@ -0,0 +1,170 @@
+
+const css = stylesheet({
+  errorMessage: {
+    color: 'red',
+  },
+  // TODO: move to Css.tsx and probably rename.
+  explanation: {
+    fontSize: fontsize.small,
+  },
+});
+
+interface ExperimentProps {
+  namespace?: string;
+}
+
+type NewExperimentFCProps = ExperimentProps & PageProps;
+
+export function NewExperimentFC(props: NewExperimentFCProps) {
+  const urlParser = new URLParser(props);
+  const { namespace, updateDialog, updateSnackbar, updateToolbar } = props;
+  const [description, setDescription] = useState<string>('');
+  const [experimentName, setExperimentName] = useState<string>('');
+  const [isbeingCreated, setIsBeingCreated] = useState<boolean>(false);
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const pipelineId = urlParser.get(QUERY_PARAMS.pipelineId);
+
+  useEffect(() => {
+    updateToolbar({
+      actions: {},
+      breadcrumbs: [{ displayName: 'Experiments', href: RoutePage.EXPERIMENTS }],
+      pageTitle: 'New experiment',
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    if (!experimentName) {
+      setErrorMessage('Experiment name is required');
+    } else {
+      setErrorMessage('');
+    }
+  }, [experimentName]);
+
+  const newExperimentMutation = useMutation((experiment: V2beta1Experiment) => {
+    return Apis.experimentServiceApiV2.createExperiment(experiment);
+  });
+
+  const create = () => {
+    let newExperiment: V2beta1Experiment = {
+      display_name: experimentName,
+      description: description,
+      namespace: namespace,
+    };
+    setIsBeingCreated(true);
+
+    newExperimentMutation.mutate(newExperiment, {
+      onSuccess: async response => {
+        let searchString = '';
+        if (pipelineId) {
+          const latestVersion = await getLatestVersion(pipelineId);
+          searchString = new URLParser(props).build({
+            [QUERY_PARAMS.experimentId]: response.experiment_id || '',
+            [QUERY_PARAMS.pipelineId]: pipelineId,
+            [QUERY_PARAMS.pipelineVersionId]: latestVersion?.pipeline_version_id || '',
+            [QUERY_PARAMS.firstRunInExperiment]: '1',
+          });
+        } else {
+          searchString = new URLParser(props).build({
+            [QUERY_PARAMS.experimentId]: response.experiment_id || '',
+            [QUERY_PARAMS.firstRunInExperiment]: '1',
+          });
+        }
+        props.history.push(RoutePage.NEW_RUN + searchString);
+        updateSnackbar({
+          autoHideDuration: 10000,
+          message: `Successfully created new Experiment: ${newExperiment.display_name}`,
+          open: true,
+        });
+      },
+      onError: async err => {
+        const errorMessage = await errorToMessage(err);
+        updateDialog({
+          buttons: [{ text: 'Dismiss' }],
+          onClose: () => setIsBeingCreated(false),
+          content: errorMessage,
+          title: 'Experiment creation failed',
```

Comment:
We might want to move this part to an useEffect to keep the mutation logic focusing on the API interaction only.